### DISCUSSION TITLE
multiple attributes for one property

### DISCUSSION THREAD

**Discussion by dietercoopman** [spatie/laravel-data#538] [Q&A]
if i use multiple attributes it breaks my code ... I want the string $email to be validated against the email rules and also want to expose my field in the api with another output name.  If I apply only one of the rules it works as expected but the combination breaks my tests.  If i combine them my validation works, but the result in the database for my creation tests email address is null

```php
<?php

namespace App\Data;

use Spatie\LaravelData\Attributes\MapOutputName;
use Spatie\LaravelData\Attributes\Validation\Email;
use Spatie\LaravelData\Data;

class AccountantData extends Data
{
    public function __construct(
        public string $title,
        #[Email, MapOutputName('account_email')]
        public string $email,
    ) {

    }
}
```

**test** 
```php
test('accountant can be created', function () {

    $response = $this->post(route('accountants.store'), [
        'title' => 'Test Accountant',
        'address' => 'Test Address',
        'email' => '1@accountant.be']);

    $this->assertDatabaseHas('accountants', ['title' => 'Test Accountant', 'email' => '1@accountant.be']);

    $response->assertStatus(201);
});
```

**method**
```php
    public function store(AccountantData $request)
    {
        return Accountant::create($request->all());
    }
```

**result in database**
```php
        "title" => "Test Accountant"
        "email" => null
```

**Comment by rubenvanassche** [comment]
`MapOutputName('account_email')` will map the email field when saving to the database. So your database field should be `account_email`.

**Reply by dietercoopman** [reply]
Hmm , i thought the output name acted like a laravel resource ... I want my database field with name x , but expose name y in my api.

In my case => 'account_email' in the database and 'email' for my api exposure.

I hoped to use a Data object for input validation + resource mapping 

Thx

**Reply by rubenvanassche** [reply]
Yeah some people expect that the property will be mapped in the db some don't not a consensus at the moment what the correct option is. In such cases, creating  a data object + resource object is most of the times the better solution IMHO.

**Reply by dietercoopman** [reply]
Ok i will use a separate resource but for me it sounds loud and clear 

MapInputName('email') on account_email => will map my Request var 'email' to db field 'account_email'
 
MapOutputName('email')] on account_email => will output email in my resource

but it might be a misinterpretation from me because it doesn't work as that :-)

thx for the replies

**Reply by rubenvanassche** [reply]
> MapOutputName('email')] on account_email => will output email in my resource

It will also map `account_email` to `email` when saving the data to a model, mapping is always ran when transforming a data object, wether it is a response, array, json string or store in a the database which will transform it to an array.
